packages/client/tests/e2e/_utils/run.ts

```diff
@@ -84,20 +91,63 @@ async function main() {
 
   console.log('🐳 Starting tests in docker')
   // tarball was created, ready to send it to docker and begin e2e tests
-  const testNames = args._.join(' ').replace(/\//g, '-')
-  await $`docker compose -f ${__dirname}/docker-compose.yml down --remove-orphans`
-  await $`docker compose -f ${__dirname}/docker-compose.yml build ${testNames}`
-  await $`docker compose -f ${__dirname}/docker-compose.yml up ${testNames}`
+  const testStepFiles = await glob('../**/_steps.ts', { cwd: __dirname })
+  let e2eTestNames = testStepFiles.map((p) => path.relative('..', path.dirname(p)))
+
+  if (args._.length > 0) {
+    e2eTestNames = e2eTestNames.filter((p) => args._.some((a) => p.includes(a)))
+  }
+
+  const dockerVolumes = [
+    `${path.resolve(__dirname, '..')}:/e2e`,
+    `${path.resolve(__dirname, '..', '..', '..')}:/client`,
+    `${path.resolve(__dirname, '..', '..', '..', '..', '..')}:/repo`,
+    `${path.resolve(__dirname, '..', '.cache')}:/root/.cache`,
```

Comment:
Not for now, put idea for later. We could pre-build a docker base image with pre-populated cache, so all installs would be faster, not only e2e.